src/transformStateWithClones.js

```diff
@@ -7,7 +7,27 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateVersions = [];
+  let cloneObj = { ...state };
+
+  for (const action of actions) {
```

Comment:
you can use a destructuring here like

` const { type, extraData, keysToRemove } = action;`